cityofaustin/vision-zero
# Remove units and people edits layers

## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/19398

## Testing

**URL to test:** 
Local

**Steps to test:**
1. Check out this branch, get a fresh replica of prod data. Apply hasura migrations and metadata.
2. Lets test a user edit to the units table:
```sql
select vin from units where id = 1;
select vin from units_cris where id = 1;
update units set vin = 'VZ EDIT' where id = 1;
select vin from units where id = 1;
```
3. Lets test a cris update to that same record, the unified record should show the VZ edit:
```sql
update units_cris set vin = 'CRIS UPDATE' where id = 1;
select vin from units where id = 1;
select vin from units_cris where id = 1;
```
4. Lets test a cris update to a non edited value, the unified record should show the new cris value:
```sql
select vin from units where id = 2;
select vin from units_cris where id = 2;
update units_cris set vin = 'CRIS UPDATE' where id = 2;
select vin from units where id = 2;
select vin from  units_cris where id = 2;
```
5. Now lets test a user null override. The unified record should show null instead of the cris value
```sql
select vin from units where id = 10;
select vin from units_cris where id = 10;
update units set vin = null where id = 10;
select vin from units where id = 10;
select vin from units_cris where id = 10;

-- extra test - reset the unified value to match the _cris schema
update units set vin = '1FMCU22XXVUC99371' where id = 10;
-- edit the cris value
update units_cris set vin = 'CRIS EDIT' where id = 10;
-- observe that the unified value is now back in sync with the cris value
select vin from units where id = 10;
```
6. Lets test a user edit to the people table:
```sql
select prsn_first_name from people where id = 1;
select prsn_first_name from people_cris where id = 1;
update people set prsn_first_name = 'VZ EDIT' where id = 1;
select prsn_first_name from people where id = 1;
```
7. Lets test a cris update to that same record, the unified record should show the VZ edit:
```sql
update people_cris set prsn_first_name = 'CRIS UPDATE' where id = 1;
select prsn_first_name from people where id = 1;
select prsn_first_name from people_cris where id = 1;
```
8. Lets test a cris update to a non edited value, the unified record should show the new cris value:
```sql
select prsn_first_name from people where id = 2;
select prsn_first_name from people_cris where id = 2;
update people_cris set prsn_first_name = 'CRIS UPDATE' where id = 2;
select prsn_first_name from people where id = 2;
select prsn_first_name from  people_cris where id = 2;
```
9. Now lets test a user null override. The unified record should show null instead of the cris value
```sql
select prsn_first_name from people where id = 198611;
select prsn_first_name from people_cris where id = 198611;
update people set prsn_first_name = null where id = 198611;
select prsn_first_name from people where id = 198611;
select prsn_first_name from people_cris where id = 198611;

-- extra test - reset the unified value to match the _cris schema
update people set prsn_first_name = 'ADAM' where id = 198611;
-- edit the cris value
update people_cris set prsn_first_name = 'CRIS EDIT' where id = 198611;
-- observe that the unified value is now back in sync with the cris value
select prsn_first_name from people where id = 198611;
```
10. Next we can test the cris import script. You can download some extract zips from the AWS S3 bucket and put them in `/vision-zero/etl/cris_import/extracts` then follow the test steps in  `cris_import/README.md` to run the cris import docker image and run the following command to process those extract zips:
```shell
./cris_import.py --csv
```
11. You can also inspect the units and people change logs and make sure you are seeing your changes there as expected
12. Test the down migrations


---

#### Ship list
